indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -752,7 +752,9 @@ TaskStatus runHashPartitionMultiPhaseParallel(TaskToolbox toolbox) throws Except
         );
 
         // This is for potential debugging in case we suspect bad estimation of cardinalities etc,
-        LOG.debug("intervalToNumShards: %s", intervalToNumShards.toString());
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("intervalToNumShards: %s", intervalToNumShards.toString());
+        }
```

Comment:
Just removing the `.toString()` would be simpler here.

```suggestion
        LOG.debug("intervalToNumShards: %s", intervalToNumShards);
```